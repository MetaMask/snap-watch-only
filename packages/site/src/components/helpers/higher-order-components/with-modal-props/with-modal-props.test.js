import configureMockStore from 'redux-mock-store';
import React from 'react';
import { renderWithProvider } from '../../../../test/lib/render-helpers';
import withModalProps from './with-modal-props';

const mockState = {
  appState: {
    modal: {
      modalState: {
        props: {
          prop1: 'prop1',
          prop2: 2,
          prop3: true,
        },
      },
    },
  },
};

describe('withModalProps', () => {
  it('should return a component wrapped with modal state props', () => {
    const TestComponent = () => <div className="test">Testing</div>;
    const WrappedComponent = withModalProps(TestComponent);
    const store = configureMockStore()(mockState);
    const { container } = renderWithProvider(
      <WrappedComponent store={store} />,
    );

    expect(container).toMatchSnapshot();
  });
});
