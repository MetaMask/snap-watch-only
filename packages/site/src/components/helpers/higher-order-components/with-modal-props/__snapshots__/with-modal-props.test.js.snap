// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withModalProps should return a component wrapped with modal state props 1`] = `
<div>
  <div
    class="test"
  >
    Testing
  </div>
</div>
`;
